apps/citizen_advocate/web/models/user.ex

```diff
@@ -65,4 +68,42 @@ defmodule CitizenAdvocate.User do
     end
   end
 
+  defp downcase_email(changeset) do
+    email = get_change(changeset, :email)
+    if is_nil(email) do
+      changeset
+    else
+      put_change(changeset, :email, String.downcase(email))
+    end
+  end
+
+  def confirmation_changeset(user, params = %{"token" => token}) do
+    user
+    |> cast(params, [])
+    |> put_change(:hashed_confirmation_token, nil)
+    |> put_change(:confirmed_at, Ecto.DateTime.utc())
+    |> validate_token(token)
+  end
+
+  alias Comeonin.Bcrypt
+
+  defp validate_token(changeset, token) do
+    if Bcrypt.checkpw(token, changeset.data.hashed_confirmation_token) do
+      changeset
+    else
+      add_error(changeset, :confirmation_token, "invalid")
+    end
+  end
+
+  def confirmation_needed_changeset(changeset) do
+    token = SecureRandom.urlsafe_base64(64)
+    hashed_token = Comeonin.Bcrypt.hashpwsalt(token)
```

Comment:
ah, since you use `Bcrypt` twice, I'd move the `alias` to the top and remove the namespace here.